source/MaterialXTest/MaterialXRender/RenderUtil.cpp

```diff
@@ -191,6 +191,10 @@ bool ShaderRenderTester::validate(const mx::FilePath optionsFilePath)
 
     setupTime.endTimer();
 
+    if (!options.enableDirectLighting)
```

Comment:
@niklasharrysson . If this is not done then we end up in a state where the direct lighgin code is emitted but the definition of the uniforms for direct lighting are not for unlit shaders. There is some inconsistency it how things are emitted I believe due to the classification of the shader graph relying on checking if the root node is explicit a "ND_surface_unit" but it's the implementation perhaps needs to be detected?